driver/controller.go

```diff
@@ -451,6 +457,56 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	return nil, status.Error(codes.Unimplemented, "")
 }
 
+// ControllerExpandVolume is called from the resizer to increase the volume size.
+func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
+	volID := req.GetVolumeId()
+
+	if len(volID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID missing in request")
+	}
+	volume, err := d.cloudscaleClient.Volumes.Get(ctx, volID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume could not retrieve existing volume: %v", err)
+	}
+
+	resizeGigaBytes, err := calculateStorageGB(req.GetCapacityRange(), volume.Type)
+	if err != nil {
+		return nil, status.Errorf(codes.OutOfRange, "ControllerExpandVolume invalid capacity range: %v", err)
+	}
+
+	log := d.log.WithFields(logrus.Fields{
+		"volume_id": req.VolumeId,
+		"method":    "controller_expand_volume",
+	})
+
+	log.Info("controller expand volume called")
+
+	if resizeGigaBytes <= volume.SizeGB {
+		log.WithFields(logrus.Fields{
+			"current_volume_size":   volume.SizeGB,
+			"requested_volume_size": resizeGigaBytes,
+		}).Info("skipping volume resize because current volume size exceeds requested volume size")
+		// even if the volume is resized independently from the control panel, we still need to resize the node fs when resize is requested
+		// in this case, the claim capacity will be resized to the volume capacity, requested capcity will be ignored to make the PV and PVC capacities consistent
+		return &csi.ControllerExpandVolumeResponse{CapacityBytes: int64(resizeGigaBytes) * GB, NodeExpansionRequired: true}, nil
```

Comment:
Is this correct to return  `int64(resizeGigaBytes) * GB` here? Or would it be better to return `volume.SizeGB * GB` instead? I'm not really sure, but I guess the user would not be able to use the volume fully if we change some code. I guess at the moment those to things should match anyway. This is more of a discussion I guess and I'm totally fine letting this how it is.

nit: capcity typo

Reply:
Good catch. I think it should use volume.SizeGB actually. Will change it.